app/presenters/v3/space_quota_presenter.rb

```diff
@@ -56,12 +56,12 @@ def space_quota
     end
 
     def filtered_visible_spaces
-      visible_spaces = if @all_spaces_visible
-                         space_quota.spaces
-                       else
-                         space_quota.spaces.select { |space| @visible_space_guids.include? space.guid }
-                       end
-      visible_spaces.map { |space| { guid: space.guid } }
+      visible_spaces_dataset = if @all_spaces_visible
+                                 space_quota.spaces_dataset
+                               else
+                                 space_quota.spaces_dataset.where(guid: @visible_space_guids)
```

Comment:
Ultimately `@visible_space_guids` comes from [`permissions.readable_space_guids`](https://github.com/cloudfoundry/cloud_controller_ng/blob/9b793086e472453382a451244eccebd49d9059ac/lib/cloud_controller/permissions.rb#L155-L157) but maybe we should instead that should be [`permissions.readable_space_guids_query`](https://github.com/cloudfoundry/cloud_controller_ng/blob/9b793086e472453382a451244eccebd49d9059ac/lib/cloud_controller/permissions.rb#L159-L165) (though we changed that recently to raise an error if the user reads globally..)